### PR TITLE
[Feature] Add conversion metrics

### PR DESCRIPTION
**Pull request by fivetran-avinash**
## PR Overview
**This PR will address the following Issue/Feature:** This is part of the add conversion metrics project

**This PR will result in the following new package version:** v0.6.0

We are changing the data type of one field in the source package that will change it in the transform, and customers bringing in the new conversion fields previously via passthrough columns will experience errors if they don't update.

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

# Feature Updates: Conversion Support!
- We have added the following source fields to each `tiktok_ads` end model:
  - `real_time_conversion`: Number of times your ad resulted in the optimization event you selected.
  - `total_purchase_value`: The total value of purchase events that occurred in your app that were recorded by your measurement partner.
  - `total_sales_lead_value`: The monetary worth or potential value assigned to a lead generated through ads.
  - `total_conversion_value`: The accumulated value of `total_purchase_value` and `total_sales_lead_value`.            

- In the event that you were already passing the above fields in via our [passthrough columns](https://github.com/fivetran/dbt_tiktok_ads?tab=readme-ov-file#passing-through-additional-metrics), the package will dynamically avoid "duplicate column" errors.
> The above new field additions are **breaking changes**  for users who were not already bringing in conversion fields via passthrough columns.

## Breaking Changes (in the `dbt_tiktok_ads_source` package)
- Update `conversion` to be an integer rather than a numeric data type, as is the expected behavior of the field. To reflect that data type change, you will need to do a `--full-refresh`.

## Under the Hood
- Created `tiktok_ads_persist_pass_through_columns` macro to ensure that the new conversion fields are backwards compatible with users who have already included them via passthrough fields.
- Added integrity and consistency validation tests within `integration_tests` folder for the transformation models (to be used by maintainers only).

## Contributors
- [Seer Interactive](https://www.seerinteractive.com/?utm_campaign=Fivetran%20%7C%20Models&utm_source=Fivetran&utm_medium=Fivetran%20Documentation)

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [NA] dbt run (if incremental models are present) && dbt test

Before marking this PR as "ready for review" the following have been applied:
- [NA] The appropriate issue has been linked, tagged, and properly assigned
- [x] All necessary documentation and version upgrades have been applied
- [x] docs were regenerated (unless this PR does not include any code or yml updates)
- [x] BuildKite integration tests are passing
- [x] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

Unfortunately the JSON issue surrounding `age_groups` in the `ad_group` model did not allow us to fully validate on dev data, so we had to use integration tests to validate nothing looks out of the ordinary.

Screenshot 2024-10-02 at 1 59 44 PM

### If you had to summarize this PR in an emoji, which would it be?
 
🕙

### CODE REVIEW

**Review by fivetran-jamie — APPROVED** [review]

File: packages.yml [hunk 1]
```diff
@@ -1,3 +1,7 @@
 packages:
-- package: fivetran/tiktok_ads_source
-  version: [">=0.5.0", "<0.6.0"]
+# - package: fivetran/tiktok_ads_source
+#   version: [">=0.6.0", "<0.7.0"]
+
+- git: https://github.com/fivetran/dbt_tiktok_ads_source.git
```
Comment (fivetran-jamie) [inline packages.yml:5]: switch

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,25 @@
...
+  - `total_sales_lead_value`: The monetary worth or potential value assigned to a lead generated through ads.
+  - `total_conversion_value`: The accumulated value of `total_purchase_value` and `total_sales_lead_value`.
+
+- In the event that you were already passing the above fields in via our [passthrough columns](https://github.com/fivetran/dbt_tiktok_ads?tab=readme-ov-file#passing-through-additional-metrics), the package will dynamically avoid "duplicate column" errors.
+> The above new field additions are **breaking changes**  for users who were not already bringing in conversion fields via passthrough columns.
+
+## Under the Hood
+- Created `tiktok_ads_persist_pass_through_columns` macro to ensure that the new conversion fields are backwards compatible with users who have already included them via passthrough fields.
+- Updated `conversion` to be an integer rather than a numeric data type in the `tiktok_ads_source` package, as is the expected behavior of the field. To reflect that data type change, **you will need to do a `--full-refresh`. This is a breaking change.**
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:16]: Will a full refresh really be required? We don't leverage any incremental logic in this package so I don't believe this needs to be called out.

File: integration_tests/tests/consistency/consistency_ad_group_report.sql [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    tags="fivetran_validations",
+    enabled=var('fivetran_validation_tests_enabled', false)
+) }}
+
+with prod as (
+    select
+        ad_group_id,
+        sum(clicks) as clicks, 
+        sum(impressions) as impressions,
+        sum(spend) as spend
```
Comment (fivetran-joemarkiewicz) [inline integration_tests/tests/consistency/consistency_ad_group_report.sql:11]: Can we include the conversion metrics in these tests (even if just commented out) so we don't forget them when we make future releases.

Comment (fivetran-joemarkiewicz) [reply integration_tests/tests/consistency/consistency_ad_group_report.sql:11]: Same comment for other consistency tests

**Review by fivetran-joemarkiewicz — APPROVED** [review]
LGTM with just a few documentation and test question/suggestions

File: CHANGELOG.md (hunk 2 above)
Comment (fivetran-jamie) [reply CHANGELOG.md:16]: Ah good point, will remove

**Pull request merged**